Tweak to defer copy (#6492)
fishtown-analytics/docs.getdbt.com: website/docs/docs/cloud/about-cloud-develop-defer.md

<Lightbox src src="/img/docs/reference/defer-diagram.png" width="50%" title="Use 'defer' to modify end-of-pipeline models by pointing to production models, instead of running everything upstream." />

When using `--defer`, dbt Cloud will follow this order of execution for resolving the `{{ ref() }}` functions.

1. If a development version of a deferred relation exists, dbt preferentially uses the development database location when resolving the reference.

> 2. If a development version does not exist, dbt uses the staging locations of parent relations based on metadata from the staging environment.

Reviewer: ```suggestion
2. If a development version doesn't exist, dbt uses the staging locations of parent relations based on metadata from the staging environment.
```

> 3. If a development version AND staging version do not exist, dbt uses the production locations of parent relations based on metadata from the production environment.

Reviewer: ```suggestion
3. If both a development and staging version doesn't exist, dbt uses the production locations of parent relations based on metadata from the production environment.
```

> **Note:** Passing the `--favor-state` flag will always resolve refs using production metadata, regardless of the presence of a development relation. This will effectively by pass step #1 above.

Reviewer: ```suggestion
**Note:** Passing the `--favor-state` flag will always resolve refs using production metadata, regardless of the presence of a development relation, skipping step #1.
```
